lib/services/consumed/autocomplete.js

```diff
@@ -0,0 +1,140 @@
+  store.subscriptions.add(disposable);
+  return disposable;
+}
+
+/**
+ * This function is just for cleaning up when either autocomplete or hydrogen is deactivating.
+ *
+ * @function disableAutocomplete
+ * @param {Store} store - The global Hydrogen store.
+ */
+export function disableAutocomplete(store: Store) {
+  // Removes the consumed function `watchEditor`
+  _autocompleteConsumer.addAutocompleteToEditor = () => {};
+
+  /*
+   * Removes disposables from all watches (leaves references inside
+   * `_autocompleteConsumer.disposables` to be disposed at the end).
+   * Autocomplete is only actually disabled on dispose of `_autocompleteConsumer.disposables`
+   */
+  for (let kernel of store.runningKernels) {
+    for (let watch of kernel.watchesStore.watches) {
+      watch.autocompleteDisposable = null;
+    }
+    kernel.watchesStore.autocompleteDisposables = new CompositeDisposable();
+  }
+
+  // Disables autocomplete, Cleans up everything, and Resets.
+  _autocompleteConsumer.disposables.dispose();
+  _autocompleteConsumer.disposables = new CompositeDisposable();
+}
+
+/**
+ * This function is for adding autocomplete capabilities to a watch.
+ *
+ * @function addAutocompleteToWatch
+ * @param {WatchesStore} watchesStore - This should always be the parent `WatchesStore` of `watch`.
+ * @param {WatchStore} watch - The watch to add autocomplete to.
+ */
+export function addAutocompleteToWatch(
+  watchesStore: WatchesStore,
+  watch: WatchStore
+) {
+  const disposable = _autocompleteConsumer.addAutocompleteToEditor(
+    watch.editor,
+    ["default", "workspace-center", "symbol-provider"]
+  );
+  if (disposable) {
+    watch.autocompleteDisposable = disposable;
+    watchesStore.autocompleteDisposables.add(disposable);
+  }
+}
+
+/**
+ * This function is for removing autocomplete capabilities from a watch.
+ *
+ * @function removeAutocompleteFromWatch
+ * @param {WatchesStore} watchesStore - This should always be the parent `WatchesStore` of `watch`.
+ * @param {WatchStore} watch - The watch to remove autocomplete from.
+ */
+export function removeAutocompleteFromWatch(
+  watchesStore: WatchesStore,
+  watch: WatchStore
+) {
+  const disposable = watch.autocompleteDisposable;
+  if (disposable) {
+    watchesStore.autocompleteDisposables.remove(disposable);
+    disposable.dispose();
+    watch.autocompleteDisposable = null;
+  }
+}
+
+/**
+ * Removes and disposes an autocomplete disposable
+ *
+ * @param {Disposable | CompositeDisposable} disposable
+ */
+export function disposeAutocompleteDisposable(
+  disposable: Disposable | CompositeDisposable
+) {
+  _autocompleteConsumer.disposables.remove(disposable);
+  disposable.dispose();
+}
+
+/**
+ * Adds a disposable as an autocomplete disposable.
+ *
+ * @param {Disposable | CompositeDisposable} disposable
+ */
+export function registerAutocompleteDisposable(
+  disposable: Disposable | CompositeDisposable
+) {
+  _autocompleteConsumer.disposables.add(disposable);
+}
```

Comment:
I think this file and it's exported functions might be better off as a class, but I wouldn't block the PR over that just wondered if you had thoughts on a class vs a set of functions.

Reply:
I could make this a class for easier naming purposes